examples/srlinux/2node-srl-ixr6-with-oc-services.pbtxt

```diff
@@ -0,0 +1,62 @@
+# This topology consists of two Nokia SR Linux IXR-6e nodes running 22.6.3 release and connected with a single link over ethernet-1/1 interfaces.
+# Each of the nodes comes up with a startup config which is sourced from 22.6.3-opeconfig.cfg.json file;
+# the startup config contains a self-signed TLS cert and has gnmi, gnoi, gribi, p4rt services enabled
+#
+# Read more about this topology at http://learn.srlinux.dev/tutorials/infrastructure/kne/srl-with-kne/
+
+name: "2-srl-ixr6"
+nodes: {
+    name: "srl1"
+    vendor: NOKIA
+    model: "ixr6e"
+    config:{
+        image: "ghcr.io/nokia/srlinux:22.6.3"
+        file: "22.6.3-opeconfig.cfg.json"
+    }
+    services:{
+        key: 22
+        value: {
+            name: "ssh"
+            inside: 22
+        }
+    }
+    services:{
+        key: 57400
```

Comment:
Ideally we could map these services and ports in the example to the openconfig "standard" ports:

```
  services: {
    key: 9337
    value: {
      name: "gnoi"
      outside: 9337
      inside: <YOUR CONTAINER gNOI PORT>
    }
  }
  services: {
    key: 9338
    value: {
      name: "gnsi"
      outside: 9338
      inside: <YOUR CONTAINER gNSI PORT>
    }
  }
  services: {
    key: 9339
    value: {
      name: "gnmi"
      outside: 9339
      inside: <YOUR CONTAINER gNMI PORT>
    }
  }
  services: {
    key: 9340
    value: {
      name: "gribi"
      outside: 9340
      inside: <YOUR CONTAINER gRIBI PORT>
    }
  }
  services: {
    key: 9559
    value: {
      name: "p4rt"
      outside: 9559
      inside: <YOUR CONTAINER P4RT PORT>
    }
  }
}
```

The outside and inside port specifications allow this remapping. The key value does not have to match the inside or outside value

Reply:
Oh neat. I will do that. I simply missed them :( 

One important question -- here you have gnoi defined over port 9337. I dont see this port to be assigned by IANA https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xml
Moreover, it says that gnoi-gnmi share the same 9339 port. Same is visible in the grpc-tunnel project that makes no distinction between the two protocols. 

Out implementation on srlinux also runs two servers on the same socket... So I wonder where does 9337 for gnoi comes from?

Reply:
added missing services config using proposed port numbers in [6a08c8c](https://github.com/openconfig/kne/pull/228/commits/6a08c8ccc9d208b446be5529f3c17b790b4bff35)

note, that gnmi and gnoi runs over the same port as per my prev comment

Reply:
Ah there was a typo is my example. The `gNxI` services can all have the same outside/inside ports and still have different service keys. When ondatra dials a service it checks for `gnoi` or `gnmi` etc.  in the service proto so it would be helpful to specify them each separately even if they point to the same ports

That should work out fine but let me know if there are any reported conflicts

Reply:
I was not able to use the same key for gnmi/gnoi. Either metallb or kne barked about key duplicates

Reply:
the key needs to be unique but the ports can be the same, ondatra only looks at the service name

Reply:
Thanks. I will have another look

Reply:
@alexmasi I think I fixed that by carving out gnoi service in cd1148b